src/shared/components/data-table/components/table-controls/components/table-actions/components/download-data/DownloadData.tsx

```diff
@@ -0,0 +1,113 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { ReactNode } from 'react';
+
+import useDataTable from 'src/shared/components/data-table/hooks/useDataTable';
+import { downloadTextAsFile } from 'src/shared/helpers/downloadAsFile';
+
+import { PrimaryButton } from 'src/shared/components/button/Button';
+
+import { TableAction } from 'src/shared/components/data-table/dataTableTypes';
+
+import styles from './DownloadData.scss';
+
+const getReactNodeText = (node: ReactNode): string => {
+  if (['string', 'number'].includes(typeof node)) {
+    return node?.toString() || '';
+  }
+
+  if (typeof node === 'object' && node && 'props' in node) {
```

Comment:
Checking if `node` is truthy is redundant here

Reply:
That is because `typeof(null)` is also `object`

Reply:
No, it'll throw `TypeError`

Reply:
Okay that makes sense. I just tried with `node && 'props' in node` (without `typeof node === 'object'`) and there was no error thrown even when it was of a different type i.e. array.